### PR TITLE
Version loading tweak

### PR DESCRIPTION
**Pull request by rgozim**
Replaced ResourceBundle omero.version with version in jar manifest
Updated gitignore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,5 +36,5 @@ gradle-local.properties
 
 # Only include main and test from src dir (for now)
 src/*
-!src/main/*
-!src/test/*
\ No newline at end of file
+!src/main/
```
Comment (sbesson) [inline .gitignore]: Is this related to the proposed change?

File: src/main/java/omero/gateway/BuildConfig.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package omero.gateway;
+
+public class BuildConfig {
+    public static String getBuildVersion(){
+        return BuildConfig.class.getPackage().getImplementationVersion();
```
Comment (sbesson) [inline src/main/java/omero/gateway/BuildConfig.java]: Looks like a very neat way to introspect a package version. For reference, the way Bio-Formats does that involves some custom code loading the manifest - see https://github.com/openmicroscopy/bioformats/blob/v6.0.0/components/formats-api/src/loci/formats/FormatTools.java#L274.

If this works, this looks definitely more compact and scalable.

File: src/main/java/omero/gateway/Gateway.java [hunk 3]
```diff
@@ -1135,8 +1134,7 @@ private ExperimenterData login(client client, LoginCredentials cred)
             ServiceFactoryPrx entryEncrypted = client.getSession();
             
             // version check
-            ResourceBundle bundle = ResourceBundle.getBundle("omero");
-            String clientVersion = bundle.getString("omero.version");
+            String clientVersion = BuildConfig.getBuildVersion();
```
Comment (sbesson) [inline src/main/java/omero/gateway/Gateway.java]: Since the `BuildConfig` class is only introspecting the package, any reason we couldn't use `self.getClass().getPackage().getImplementationVersion();`

Comment (rgozim) [reply src/main/java/omero/gateway/Gateway.java]: I opted to stay with the BuildConfig class for now as it keeps a single source of truth. We can swap out the implementation of how the version is obtained, but not how to change the calling code. 

BuildConfig is also a pattern used on [Android](https://developer.android.com/studio/build/gradle-tips#simplify-app-development) and works very well with Gradle. It's hard to vouch for it, but it's something that has stood up well as the build system became more dynamic.

**Comment by pwalczysko** [comment]
Tested with this PR, building OMERO.insight as well as OMERO.web.

With OMERO.insight, I can connect to eel (a 5.5 server). I cannot connect to nightshade (5.4 server).

Screen Shot 2019-03-21 at 11 05 49


With OMERO.web, I can connect to my localhost (5.5 server).

**Comment by jburel** [comment]
we used have a special handling of ``@OMERO_Version`` something similar is needed now for the SNAPSHOT case
The check needs to be adjusted to support snapshot,

File: .gitignore (hunk 1 above)
Comment (jburel) [reply .gitignore]: not really
This could be added to another PR. This is mainly to ensure that only the generated classes are excluded

Comment (jburel) [reply .gitignore]: Reading the exclusion example from https://git-scm.com/docs/gitignore
This seems to be the correct pattern to not exclude ``src/main``

**Comment by jburel** [comment]
merging for m5

File: .gitignore (hunk 1 above)
Comment (mtbc) [reply .gitignore]: > It is not possible to re-include a file if a parent directory of that file is excluded.

Comment (mtbc) [reply .gitignore]: (i.e. I think this fix might indeed successfully work around the previous problem)
